velox/common/caching/SsdFile.cpp

```diff
@@ -677,14 +653,16 @@ void SsdFile::deleteCheckpoint(bool keepLog) {
     }
   }
 
-  const auto checkpointPath = getCheckpointFilePath();
-  const auto checkpointRc = ::unlink(checkpointPath.c_str());
-  if (checkpointRc != 0) {
-    VELOX_SSD_CACHE_LOG(ERROR)
-        << "Error in deleting checkpoint: " << checkpointRc;
-  }
-  if (checkpointRc != 0) {
-    ++stats_.deleteCheckpointErrors;
+  if (checkpointWriteFile_ != nullptr) {
```

Comment:
We don't need to maintain checkpointWriteFile_? It can be temp object while we write checkpoint? Thanks!

Reply:
Right we don't need to maintain this. But the idea is to get a hold of the checkpointwritefile so we can disable cow for all files at once: https://github.com/facebookincubator/velox/blob/ddda3ab7b7863e89d518bb99e3fb2adcc4c6149d/velox/common/caching/SsdFile.cpp#L900-L912

Otherwise we need to disable it everytime we do checkpoint?